src/main/java/com/slavalapin/ifmo/web/servlets/StringCalculator.java

```diff
@@ -0,0 +1,144 @@
+package com.slavalapin.ifmo.web.servlets;
+
+import java.io.*;
+import java.util.*;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet(
+        name = "StringCalculator",
+        urlPatterns = "/calc"
+)
+
+public class StringCalculator extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest requestCalculation, HttpServletResponse responseCalculation)
+            throws ServletException, IOException {
+
+        final Map<String, String[]> requestParameters = requestCalculation.getParameterMap();
+        String expressionOriginal = requestParameters.get("equation")[0];
+
+        ArrayDeque<String> expressionReady = mapVariables(toPolishNotation(expressionOriginal), requestParameters);
+        int result = compute(expressionReady);
+
+        //StringBuilder result = new StringBuilder();
+        /*while (!expressionReady.isEmpty()) {
+            result.append(expressionReady.poll());
+            result.append(" ");
+        }*/
+
+
+        PrintWriter responseBodyWriter = responseCalculation.getWriter();
+        responseBodyWriter.print(result/*.toString()*/);
+        responseBodyWriter.flush();
+        responseBodyWriter.close();
+    }
+
+
+    private int compute(ArrayDeque<String> expression) {
+       Stack<Integer> calculationStack = new Stack<>();
+       while (!expression.isEmpty()) {
+           String step = expression.poll();
+           if(Character.isDigit(step.charAt(0)))
+               calculationStack.push(Integer.parseInt(step));
+           else {
+               char symbol = step.charAt(0);
+               int rightOperand = calculationStack.pop();
+               int leftOperand = calculationStack.pop();
+               int resultIntermediate;
+               switch (symbol) {
+                   case '+':
+                       resultIntermediate = rightOperand + leftOperand;
+                       break;
+                   case '-':
+                       resultIntermediate = leftOperand - rightOperand;
+                       break;
+                   case '*':
+                       resultIntermediate = rightOperand * leftOperand;
+                       break;
+                   case '/':
+                       resultIntermediate = leftOperand / rightOperand;
+                       break;
+                   default:
+                       throw new java.lang.IllegalStateException("Unexpected value: " + step);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'java.lang.IllegalStateException' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=5345481)